### PR TITLE
JS linting in Travis build 

### LINKED ISSUE #482: Add JS linting to Travis builds
Use `eslint` in the Travis build to: 
- fail the build on JS errors 
- display linter warnings 
- possibly limit the number of allowed warnings

### PR DESCRIPTION
**Pull request by ShayNehmad**
# Feature / Fixes
Fixes #482

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,12 +9,20 @@ cache: pip
 python:
 - 3.7
 
+os: linux
```
Comment (VakarisZ) [inline .travis.yml]: what does this line mean?

File: .travis.yml [hunk 2]
```diff
@@ -9,12 +9,20 @@ cache: pip
...
+
 install:
+# Python
 - pip install -r monkey/monkey_island/requirements.txt  # for unit tests
 - pip install flake8 pytest dlint  # for next stages
 - pip install -r monkey/infection_monkey/requirements_linux.txt  # for unit tests
 
 before_script:
+# Set the server config to `testing`, for the UTs to use mongomaock and pass.
```
Comment (VakarisZ) [inline .travis.yml]: typo, but dont fix if it's the only problem

File: .travis.yml (hunk 1 above)
Comment (ShayNehmad) [reply .travis.yml]: That the build machine will be a linux machine. 
https://docs.travis-ci.com/user/multi-os/
In future, I'd like to add Windows as well, but TBH the UTs should be OS agnostic anyway...

File: .travis.yml (hunk 2 above)
Comment (ShayNehmad) [reply .travis.yml]: ad1e85f8

**Pull request merged**